cmd/changelog-parser/main.go

```diff
@@ -79,73 +79,104 @@ func fetchChangelog(
 	return string(changelog), nil
 }
 
-func collectChangelogs(repoConfig repositories.Config, httpClient *http.Client) (
-	[]*changelogPkg.VersionChangelog,
-	error,
-) {
+func componentFromRepo(httpClient *http.Client, repo repositories.Repository) (template.Component, error) {
+	component := template.Component{
+		Repo: repo.Name,
+	}
+
 	var changelogs []*changelogPkg.VersionChangelog
-	for _, category := range repoConfig.Section.Categories {
-		log.Printf("Processing category: %s", category.Name)
-		for _, repo := range category.Repos {
-			log.Printf("- Processing repo: %s", repo.Name)
 
-			if repo.Version == "" {
-				// TODO: This should be somehow transformed from repo url
-				version, err := latestVersionToExactVersion(httpClient, "github", repo.Name)
-				if err != nil {
-					return nil, err
-				}
+	if repo.Version == "" {
+		// TODO: This should be somehow transformed from repo url
+		version, err := latestVersionToExactVersion(httpClient, "github", repo.Name)
+		if err != nil {
+			return component, err
+		}
 
-				repo.Version = version
-			}
+		repo.Version = version
+	}
 
-			availableVersions, err := github.GetAvailableReleases(httpClient, repo.Name)
-			if err != nil {
-				return nil, err
-			}
+	// Repo version is the linked component release version
+	component.ReleaseName = repo.Version
+
+	availableVersions, err := github.GetAvailableReleases(httpClient, repo.Name)
+	if err != nil {
+		return component, err
+	}
 
-			relevantVersions, err := version.GetRelevantVersions(
-				availableVersions,
-				repo.AfterVersion,
-				repo.Version,
+	relevantVersions, err := version.GetRelevantVersions(
+		availableVersions,
+		repo.AfterVersion,
+		repo.Version,
+	)
+	if err != nil {
+		return component, err
+	}
+
+	log.Printf("  Relevant versions: [%s]", strings.Join(relevantVersions, ", "))
+
+	// TODO: This should be somehow transformed from repo url
+	completeChangelog, err := fetchChangelog(httpClient, "github", repo.Name, "master")
+	if err != nil {
+		return component, err
+	}
+
+	// XXX: This still doesn't address releases and how we include that data in yet.
+	for _, relevantVersion := range relevantVersions {
+		log.Printf("  Extracting changelog data from %s...", relevantVersion)
+		versionChangelog, err := extractVersionChangeLog(
+			repo.Name,
+			relevantVersion,
+			completeChangelog,
+		)
+		if err != nil {
+			return component, err
+		}
+
+		if versionChangelog == nil {
+			log.Printf(
+				"  CHANGELOG not found for %s@%s",
+				repo.Name,
+				relevantVersion,
 			)
-			if err != nil {
-				return nil, err
-			}
+			continue
+		}
+
+		// If this changelog is for the suite release pinned version, save the release
+		// date. Since VersionChangelogs are stripped of `v` prefix, we need it back for
+		// this comparison.
+		if "v"+versionChangelog.Version == component.ReleaseName {
```

Comment:
I wonder if `versionChangelog.Version` and `component.ReleaseName` are each always guaranteed to either have or not have the v-prefix. If there's no guarantee then to avoid edge-cases it'd be useful to compare using `strings.TrimPrefix(versionChangelog.Version, "v") == strings.TrimPrefix(component.ReleaseName, "v")`